packages/@aws-cdk/aws-events-targets/lib/state-machine.ts

```diff
@@ -13,13 +13,29 @@ export interface SfnStateMachineProps {
    * @default the entire EventBridge event
    */
   readonly input?: events.RuleTargetInput;
+
+  /**
+   * The IAM role to be assumed to execute the State Machine
+   *
+   * @default - a new role will be created
+   */
+  readonly role?: iam.IRole;
 }
 
 /**
  * Use a StepFunctions state machine as a target for Amazon EventBridge rules.
  */
 export class SfnStateMachine implements events.IRuleTarget {
+  private readonly role: iam.IRole;
+
   constructor(public readonly machine: sfn.IStateMachine, private readonly props: SfnStateMachineProps = {}) {
+    if (props.role) {
+      this.role = props.role;
+      this.role.grant(new iam.ServicePrincipal('events.amazonaws.com'));
+    } else {
+      this.role = singletonEventRole(machine, []);
+    }
+    machine.grantStartExecution(this.role);
```

Comment:
how about something like this to simplify

```suggestion
    if (props.role) {
      props.role.grant(new iam.ServicePrincipal('events.amazonaws.com'));
    }
    machine.grantStartExecution(props.role ?? singletonEventRole(machine, []));
```

Reply:
Same as above. We need to have a defined role on the instance.

Reply:
sure. it can still be simplified though I think?

```suggestion
    if (props.role) {
      props.role.grant(new iam.ServicePrincipal('events.amazonaws.com'));
    }
    this.role = props.role ?? singletonEventRole(machine, []));
    machine.grantStartExecution(this.role);
```